lib/srv/ctx.go

```diff
@@ -479,8 +484,53 @@ func (c *ServerContext) takeClosers() []io.Closer {
 }
 
 func (c *ServerContext) Close() error {
+	// When the ServerContext (connection) is closed, emit "session.data" event
+	// containing how much data was transmitted and received over the net.Conn.
+	defer func() {
```

Comment:
this should be a separate method of `serverContext` that should return if the underlying connection is not tracking(does not have Stater` interface. This way it will be cleaner:

```go
// Stater is extension interface of the net.Conn
// for implementations that track connection statistics
type Stater interface {
   // Stat returns tx, rx data
   Stat() (int64, int64)
}

Close() error {
   if stats, ok := c.Conn.(utils.Stater); ok {
         defer c.reportStats(stats)
   }
```